File: test/fixtures/vulnerable.go
// Intentionally vulnerable Go code for testing SecureCodeAnalyzer.
package main

import (
	"database/sql"
	"fmt"
	"math/rand"
	"net/http"
	"os"
	"os/exec"
)

// SQL Injection — fmt.Sprintf in Query
func getUser(db *sql.DB, name string) {
	db.Query(fmt.Sprintf("SELECT * FROM users WHERE name='%s'", name))
}

// Hardcoded secrets
var password = "super_secret_go_password"
var api_key = "sk-abcdefghijklmnopqrst"

// Insecure random — math/rand already imported above
func generateToken() int {
	return rand.Intn(999999)
}

// Open redirect
func handleRedirect(w http.ResponseWriter, r *http.Request) {
	http.Redirect(w, r, r.URL.Query().Get("url"), 302)
}

// Path traversal — os.Open with request data
func readFile(w http.ResponseWriter, r *http.Request) {
	f, _ := os.Open(r.URL.Query().Get("path"))
	defer f.Close()
}

// Command injection — exec.Command with request data
func runCommand(w http.ResponseWriter, r *http.Request) {
	cmd := exec.Command(r.URL.Query().Get("cmd"))
	cmd.Run()
}

// XSS — unescaped HTML in response
func renderHTML(w http.ResponseWriter, r *http.Request) {
	fmt.Fprintf(w, "<div>%s</div>", r.URL.Query().Get("name"))
}

File: test/fixtures/vulnerable.java
/**
 * Intentionally vulnerable Java code for testing SecureCodeAnalyzer.
 */
import java.sql.*;
import java.io.*;
import java.util.Random;
import javax.servlet.http.*;

public class VulnerableApp {

    // SQL Injection — string concatenation in executeQuery
    void getUser(Connection conn, String name) throws Exception {
        Statement stmt = conn.createStatement();
        stmt.executeQuery("SELECT * FROM users WHERE name='" + name + "'");
    }

    // Hardcoded secrets
    String password = "database_password_123";
    String api_key = "AKIAIOSFODNN7EXAMPLEQ";

    // Insecure random
    void generateToken() {
        Random rand = new Random();
        int token = rand.nextInt();
    }

    // Path traversal — user input in File constructor
    void readFile(HttpServletRequest request) throws Exception {
        File f = new File(request.getParameter("path"));
        FileInputStream fis = new FileInputStream(f);
    }

    // XSS — unescaped HTML output
    void respond(HttpServletResponse response, String userInput) throws Exception {
        response.getWriter().write("<div>" + userInput + "</div>");
    }

    // CSRF disabled
    void configureSecurity(HttpSecurity http) throws Exception {
        http.csrf().disable();
    }

    // Open redirect
    void handleRedirect(HttpServletRequest request, HttpServletResponse response) throws Exception {
        response.sendRedirect(request.getParameter("url"));
    }

    // Command injection via Runtime.exec
    void execute(String cmd) throws Exception {
        Runtime.getRuntime().exec(cmd);
    }
}
